code/messaging/src/main/java/com/adobe/marketing/mobile/aepcomposeui/components/AepTextComposable.kt

```diff
@@ -0,0 +1,163 @@
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.aepcomposeui.components
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.adobe.marketing.mobile.aepcomposeui.style.AepTextStyle
+import com.adobe.marketing.mobile.aepcomposeui.style.merge
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepColor
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepFont
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepText
+import com.adobe.marketing.mobile.aepcomposeui.utils.UIUtils.getColor
+
+/**
+ * A composable function that displays a text element with customizable properties.
+ *
+ * @param defaultStyle The default [AepTextStyle] to be applied to the text element.
+ * @param overriddenStyle The [AepTextStyle] provided by the app that overrides the default text style.
+ */
+@Composable
+internal fun AepText.Composable(
+    defaultStyle: AepTextStyle? = null,
+    overriddenStyle: AepTextStyle? = null
+) {
+    // Convert server color from string to Color object
+    val textColor = color?.getColor()
+
+    // Convert server text alignment from string to TextAlign object
+    val textAlign = when (align?.lowercase()) {
+        "left" -> TextAlign.Left
+        "center" -> TextAlign.Center
+        "right" -> TextAlign.Right
+        else -> null
+    }
+
+    // Convert server font properties from string to respective font objects
+    // TODO map font string to FontFamily
+    // val fontFamily = null
+    val fontSize = (font?.size)?.sp
+    val fontWeight = when (font?.weight?.lowercase()) {
+        "bold" -> FontWeight.Bold
+        else -> null
+    }
+    val fontStyle = if (font?.style?.contains("italic") == true) {
+        FontStyle.Italic
+    } else {
+        null
+    }
+
+    // Merge all text styles together
+    val mergedAepTextStyle =
```

Comment:
Would it be possible to have the composable provide the `default` and `preferred` style and do 
```
val mergedTextStyle = defaultStyle.merge(overriddenStyle)
```
and the `merge` can internally merge only once like
```
fun AepTextStyle.merge(other) {
    val textStyle = TextStyle(
        color = this.textStyle?.color ?: other.textStyle?.color ?: fallbackTextStyle.color,
        fontSize = this.textStyle?.fontSize ?: other.textStyle?.fontSize ?: fallbackTextStyle.fontSize,
        fontWeight = this.textStyle?.fontWeight ?: other.textStyle?.fontWeight ?: fallbackTextStyle.fontWeight,
        fontStyle = this.textStyle?.fontStyle ?: other.textStyle?.fontStyle ?: fallbackTextStyle.fontStyle,
    )

   return AepTextStyle(textStyle = textStyle....)
  
}
```

Reply:
Moved merge to `AepTextStyle` and `AepButtonStyle` class as discussed